shopBuddy/shopBuddy/ShoppingList+Persistent.m

```diff
@@ -0,0 +1,120 @@
+//
+//  ShoppingList+Persistent.m
+//  shopBuddy
+//
+//  Created by Rachna Gupta on 7/21/22.
+//
+
+#import "ShoppingList+Persistent.h"
+#import "Item.h"
+#import "ShoppingList.h"
+#import "Parse/Parse.h"
+#import "Item+Persistent.h"
+#import <objc/runtime.h>
+
+@implementation ShoppingList (Persistent)
+
++ (void) _createObject:(PFObject *)object {
+    [object saveInBackground];
+}
+
+- (NSString *)objectID {
+    return objc_getAssociatedObject(self, @selector(objectID));
+}
+
+- (void)setObjectID:(NSString *)new_objectID {
+    objc_setAssociatedObject(self, @selector(objectID), new_objectID, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+}
+
+- (void) _updateSavedListWithNewItem: (PFObject *)item {
+    PFQuery *query = [PFQuery queryWithClassName:@"ShoppingList"];
+    [query getObjectInBackgroundWithId:self.objectID block:^(PFObject *list_object, NSError *error) {
+        if (!error) {
+            NSMutableArray *previous_items = [NSMutableArray arrayWithArray:list_object[@"items"]];
+            [previous_items addObject:item];
+            list_object[@"items"] = [NSArray arrayWithArray:previous_items];
+            [list_object saveInBackground];
+        } else {
+            NSLog(@"%@",error);
+        }
+    }];
+}
+
+//Used to create new lists
++ (void) createEmptyList:(NSString *)store_name withCompletion:(void(^)(ShoppingList *new_list))completion {
+    NSDictionary *dict = @{ @"user" : [PFUser currentUser], @"items" : [NSArray new], @"store_name" : store_name};
+    PFObject *new_object = [PFObject objectWithClassName:@"ShoppingList" dictionary:dict];
+    [ShoppingList _createObject:new_object];
+    ShoppingList *const newList = [ShoppingList _hydrateShoppingListFromPFObject:new_object];
+    completion(newList);
+    
+}
+
++ (void) addItemToList: (Item *)item withList: (ShoppingList *)list {
+    NSMutableArray *const mutable_items = [NSMutableArray arrayWithArray:list.items];
+    [mutable_items addObject:item];
+    NSArray *array = [mutable_items copy];
```

Comment:
You don't need this, `NSMutableArray` extends `NSArray` so you can use `mutable_items ` directly.